src/wp-includes/pluggable.php

```diff
@@ -2845,6 +2845,7 @@ function get_avatar( $id_or_email, $size = 96, $default_value = '', $alt = '', $
 			'loading'       => null,
 			'fetchpriority' => null,
 			'extra_attr'    => '',
+			'decoding'      => 'async',
```

Comment:
The default should be `null` (unset), as is for the other loading optimization attributes. Otherwise this partially reverts the improvement from 58892.

Reply:
@felixarntz, it appears that the `get_avatar` function does not invoke any separate function to generate the loading optimization attributes. Instead, it constructs this code within the same function. Please review the code at this location: https://github.com/WordPress/wordpress-develop/blob/6.3/src/wp-includes/pluggable.php#L2938-L2946.

If we provide `null` as an argument, it should now work for the `get_avatar` function within the Block Avatar block.

Please let me know if I have misunderstood this.

Reply:
@mukeshpanchal27 
> it appears that the `get_avatar` function does not invoke any separate function to generate the loading optimization attributes.

It does, see line 2869. That's why we should not set `async` here, it needs to come from that function call.

Reply:
Hello @pereirinha and @felixarntz, According to the information provided in the WordPress core file at https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/pluggable.php#L2868, it appears that the context for `get_avatar` is not `the_content`. As a result, it return an empty array for both the 'Avatar' and 'Author' editor blocks.

Could you please confirm whether this is working as expected on your end?

Reply:
@mukeshpanchal27 That is okay: Images in blocks that are part of `the_content` are intentionally skipped, as they should only be considered later as part of the whole content blob (with `the_content` context).